tianshou/data/batch.py

```diff
@@ -639,7 +601,40 @@ def __cat(self,
 
     def cat_(self,
              batches: Union['Batch', List[Union[dict, 'Batch']]]) -> None:
-        return self.__cat(batches)
+        """Concatenate a list of (or one) :class:`~tianshou.data.Batch` objects
+        into current batch.
+
+        If x=Batch(a=Batch(a=np.random.randn(3, 4)), b=np.random.randn(3, 4)),
+        y = Batch(a=Batch(a=Batch()), b=np.random.randn(3, 4)), if we want to
+        concatenate x and y, we want to pad y.a.a with zeros. Without ``lens``
```

Comment:
`without ``lens``
        as a hint, when we concatenate x.a and y.a, we would not be able to
        know how to pad y.a. So we first compute the variable ``lens`` to the
        internal method ``Batch.__cat`` a hint.`

Not relevant here.

Reply:
Indeed this should not appear in the doc string of ``cat_``, which can be seen by users. Moved to the doc string of ``__cat`` in f64faf5 .